.github/actions/update-versions/update-versions.mjs

```diff
@@ -0,0 +1,69 @@
+import fs from 'fs';
+import path from 'path';
+
+const args = process.argv.splice(2);
+const versionsPath = path.resolve(process.cwd(), args[1]);
+const packagePath = path.resolve(process.cwd(), args[3]);
+
+const getMajorVersion = version => version.split('.')[0];
+
+if (!fs.existsSync(versionsPath)) {
+  throw new Error(`versions.json could not be found at ${versionsPath}`);
+}
+
+if (!fs.existsSync(packagePath)) {
+  throw new Error(`package.json could not be found at ${packagePath}`);
+}
+
+const packageJSON = JSON.parse(fs.readFileSync(packagePath));
+const versionsJSON = JSON.parse(fs.readFileSync(versionsPath));
+
+const currentMajorVersion = getMajorVersion(packageJSON.version);
+const allDependencies = { ...packageJSON.dependencies, ...packageJSON.devDependencies };
+const hasVersionEntry = versionsJSON.find(
+  v => 'version' in v && getMajorVersion(v.version) === currentMajorVersion,
+);
+const interestingKeywords = ['@swisspost', '@angular/core', 'react', 'bootstrap'];
+
+let updatedVersions;
+if (hasVersionEntry) {
+  // Update the existing entry
+  updatedVersions = versionsJSON.map(version => {
+    if (!('version' in version) || getMajorVersion(version.version) !== currentMajorVersion)
+      return version;
+
+    const updatedVersion = {
+      ...version,
+      version: packageJSON.version,
+      dependencies:
+        'dependencies' in version
+          ? Object.keys(version.dependencies).reduce((acc, key) => {
+              acc[key] = allDependencies[key].replace('workspace:', '');
+              return acc;
+            }, {})
+          : null,
+    };
+
+    return updatedVersion;
+  });
+} else {
+  console.log(allDependencies);
+  // Add a new entry and sort based on version number
+  updatedVersions = [
+    ...versionsJSON,
+    {
+      title: `Version ${currentMajorVersion}`,
+      version: packageJSON.version,
+      description: packageJSON.description,
+      url: `https://design-system.post.ch/v${currentMajorVersion}`,
+      dependencies: Object.keys(allDependencies)
+        .filter(k => interestingKeywords.some(i => k.includes(i)))
+        .reduce((acc, key) => {
+          acc[key] = allDependencies[key];
+          return acc;
+        }, {}),
+    },
+  ].sort((a, b) => (a.version > b.version ? -1 : 1));
+}
+
+fs.writeFileSync(versionsPath, JSON.stringify(updatedVersions, null, 2));
```

Comment:
The const versionJSON is allready a copy of the version file.
So why you not just search and replace what you really need to change and write the edited versionJSON to the file afterwards?

Please read at least the comments about the 'workspace' keywords!

```mjs
if (hasVersionEntry) {
  // Update the existing entry
  const current = versionJSON.find(version => 'version' in version && version.version === currentMajorVersion);
  current.version = packageJSON.version;
  current.dependencies = 'dependencies' in version ?
    // The 'workspace' keyword could be removed before from the 'allDependencies' variable, because we probably don't want those keyords anywhere in the version.json
    Object.keys(version.dependencies).reduce((acc, key) => Object.assign(acc, { [key]: allDependencies[key].replace('workspace:', '') }), {})
   : null;
} else {
  // Add a new entry and sort based on version number
  versionJSON.push({
    title: `Version ${currentMajorVersion}`,
    version: packageJSON.version,
    description: packageJSON.description,
    url: `https://design-system.post.ch/v${currentMajorVersion}`,
    dependencies: Object.keys(allDependencies)
        .filter(k => interestingKeywords.some(i => k.includes(i)))
        // If you do not replace 'workspace' keyword in allDependencies[key] here, it will probably show up in the versions.json output
        .reduce((acc, key) => Object.assign(acc, { [key]: allDependencies[key] }), {})
  })
  .sort((a, b) => (a.version > b.version ? -1 : 1));

  // You probably want to update the previous current version here.
}

fs.writeFileSync(versionsPath, JSON.stringify(versionJSON, null, 2));
```

Reply:
Update the logic to also update title and url on other versions if a new one gets added (with latest major detection).

Reply:
Not really what I meant, but as long as it's running, I'm fine with it.